source/KeyHandlers.js

```diff
@@ -269,7 +269,13 @@ var keyHandlers = {
         }
 
         // If contains an inline element don't delete on first line, instead preventDefault and insert ZWS to keep tags
-        else if ( self.hasFormat( 'b', null, range ) || self.hasFormat( 'i', null, range ) || self.hasFormat( 'u', null, range ) || self.hasFormat( 'span', null, range )) {
+        // Make sure the cursor is not at the start of line
+        else if ( (range.endOffset != 0) &&
```

Comment:
@JasaNiklanovic why is this the right fix?

I'm particularly worried that it might break something else.

Reply:
Maybe this check should be merged with the one on line 280 instead, where we are already looking at the length of the text.

Reply:
Also make sure that fn/backspace works in all cases, this will clash with this fix.

Reply:
@ConradIrwin regarding your suggestion "Maybe this check should be merged with the one on line 280 instead, where we are already looking at the length of the text."

Problem i see here is in this particular case where the cursor is at the beginning of the line, the else if (hasFormat('b') ... will still fire but the actual code inside that else if will not (when i merge the check on 280); problem is that in this case i want the next else if to fire, "If at beginning of block, merge with previous". Otherwise we will not be getting this. I propose we keep this as is, unless there is something else that is problematic here?

Reply:
Note i did add tests for the delete handler, as discussed.
https://github.com/superhuman/superhuman/pull/7021